kb/deadline-exceeded-troubleshooting.md

```diff
@@ -0,0 +1,116 @@
+---
+slug: deadline-exceeded-troubleshooting
+title: DeadlineExceeded error troubleshooting
+tags:
+  - kb-article
+date: 2022-11-09T00:00:00Z
+---
+
+All client-side requests made to the [Temporal Cluster](/concepts/what-is-a-temporal-cluster) are [gRPC requests](https://grpc.io/docs/what-is-grpc/core-concepts/#deadlines).
+Sometimes, when these requests can't be completed, you'll see this particular error message: `Context: deadline exceeded`.
+Network interruptions, timeouts, server overload, and Query errors are some of the reasons that can cause this error.
+
+The following sections discuss the nature of this error and how to troubleshoot it.
+
+#### Check Frontend Service logs
+
+:::note
+
+Cloud users cannot access some of the logs needed to diagnose the source of the error.
+
+If you're using Temporal Cloud, check your Workflow History for error messages and create a [support ticket](https://support.temporal.io/) with Temporal.
+Provide the full error message in your ticket.
+
+:::
+
+[Frontend Service](/clusters/#frontend-service) logs can show which parts of the Cluster aren't working.
```

Comment:
I would maybe say there that in order for this error to appear, there is a frontend service pod/container up and running, meaning that if frontend service was down user would not get this client error (deadline exceeded) but a different error for example:


tctl cl h
Error: Unable to get "temporal.api.workflowservice.v1.WorkflowService" health check status.
Error Details: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:7233: connect: connection refused"